src/OeUI/OemrUI.php

```diff
@@ -393,39 +393,74 @@ private function headerExpandJs($arrFiles = array())
     private function headerActionJs($arrAction = array())
     {
         $arrAction = $this->arrAction;
-        $action_top_js = <<<SHWTOP
-        <script>
-        $(function () {
-            $('#show_hide').click(function () {
-                var elementTitle = '';
-SHWTOP;
-        echo $action_top_js . "\r\n";
+        $page = str_replace(" ", "", $this->heading);
 
+        // Build the labels for when the icon is moused-over
+        $labels = "";
         if ($arrAction[0] == 'search') {
-            echo "var showTitle = " .  xlj('Click to show search') . "\r\n;";
-            echo "var hideTitle = " . xlj('Click to hide search') . "\r\n;";
+            $labels .= "var showTitle = " .  xlj('Click to show search') . "\r\n;";
+            $labels .= "var hideTitle = " . xlj('Click to hide search') . "\r\n;";
         } elseif ($arrAction[0] == 'reveal' || $arrAction[0] == 'conceal') {
-            echo "var hideTitle = " .  xlj('Click to Hide') . "\r\n;";
-            echo "var showTitle = " . xlj('Click to Show') . "\r\n;";
+            $labels .= "var hideTitle = " .  xlj('Click to Hide') . "\r\n;";
+            $labels .= "var showTitle = " . xlj('Click to Show') . "\r\n;";
         }
 
+        // Build the classes for which icon to display whien hiding, showing, etc.
+        $actionClasses = "";
         if ($arrAction[0] == 'search') {
-            echo "$(this).toggleClass('fa-search-plus fa-search-minus'); \r\n";
+            $actionClasses .= "var showActionClass = 'fa-search-plus'; \r\n";
+            $actionClasses .= "var hideActionClass = 'fa-search-minus'; \r\n";
         } elseif ($arrAction[0] == 'reveal') {
-            echo "$(this).toggleClass('fa-eye fa-eye-slash'); \r\n";
+            $actionClasses .= "var showActionClass = 'fa-eye'; \r\n";
+            $actionClasses .= "var hideActionClass = 'fa-eye-slash'; \r\n";
         } elseif ($arrAction[0] == 'conceal') {
-            echo "$(this).toggleClass('fa-eye-slash fa-eye'); \r\n";
+            $actionClasses .= "var showActionClass = 'fa-eye-slash'; \r\n";
+            $actionClasses .= "var hideActionClass = 'fa-eye'; \r\n";
         }
 
+        $action_top_js = <<<SHWTOP
+        <script>
+        $(function () {
+            $labels
+            $actionClasses
+            $('#show_hide').click(function () {
+                var elementTitle = '';
+SHWTOP;
```

Comment:
This is an interesting approach:)

Reply:
@sjpadgett Jerry, the original code is a bit wack-a-doodle, so I just hacked this in there. We should really pass in the ids of the clickable elements and the hide-able elements. This code depends on the id of the clickable element being "#show_hide" and there can only be one per page (since element ids are required to be unique), and this code would show/hide all the "hideaway" elements on a page, not just the one associated with the button. I like the concept of having a generic UI generator component, but I think this particular component can lead to bugs in the future.

Reply:
Yeah there's a few places in core through the several passes to update styling, dup ids and classes were implemented.
This helper class was meant to get removed during 6.0 but was overlooked or nobody understood its implementation. Mainly the before/after div stuff.
Still this works for me and if you are comfortable, i'll bring in.

Reply:
Yep, it’s weird. I pulled on my hair for about an hour to figure it out. But yeah, it works so I’m fine with bringing it in. Maybe I can remove it, or at least make it easier to understand and use for the next version since I kinda understand what it’s doing.

Reply:
As far as i'm concerned these custom classes are better served implemented in other ways. I don't have the cycles right now and it's working for 6.0 so, later is fine.
Bringing in. 
@bradymiller This can goto release branch also.

Thanks Ken, this is very useful imo.